integration_tests/integration_tests/tests/dbt_project.py

```diff
@@ -98,15 +142,32 @@ def test(
                 YAML().dump(props_yaml, props_file)
                 relative_props_path = Path(props_file.name).relative_to(PATH)
                 self.dbt_runner.test(select=str(relative_props_path))
-        return self._read_test_result(test_id)
+
+        if multiple_results:
+            return self._read_test_results(test_id)
+        else:
+            return self._read_single_test_result(test_id)
 
     def seed(self, data: List[dict], table_name: str):
         return DbtDataSeeder(self.dbt_runner).seed(data, table_name)
 
-    def _read_test_result(self, table_name: str) -> Dict[str, Any]:
-        return self.read_table(
+    def _read_test_results(self, table_name: str) -> List[Dict[str, Any]]:
+        test_execution_id = self.read_table(
```

Comment:
1. Can we maybe do this with a `JOIN` rather than two separate queries?
2. I think this is problematic as we add more tests because it'll be much more likely that tests will run in parallel and therefore this might return the results of a different test. I think we need to think about a way to extract the invocation id and then filter by it.

Reply:
I'll change to a JOIN (or a subquery)
Re. the parallel execution - I think the table name we filter on will be different, no?